### PR TITLE
Scripts simple delay

### PR DESCRIPTION
**Pull request by hunchback**
Extended simple.sh (atop @safchain 's bridge related commit) so enable:

```
./simple.sh start                               # invocation using default subnets
./simple.sh start --delay                       # injection of netem delays
./simple.sh ping                                # ping: vm1-->vm2 and vm2-->vm1
```

### CODE REVIEW

**Comment by hunchback** [comment]
I notice the following failing - but was not sure why:

```
12:26:31.626 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:32.667 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:33.716 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:34.756 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:35.797 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:36.845 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:37.248 2018-03-09T13:26:37.183+0100	DEBUG	flow/table.go:231 (*Table).update	dpdk2: flow table update: 1520598387183, 1520598397183
12:26:37.282 2018-03-09T13:26:37.218+0100	DEBUG	flow/table.go:231 (*Table).update	dpdk2: flow table update: 1520598387217, 1520598397217
12:26:37.892 Gremlin: G.Context(1520598366543)
12:26:38.431 Gremlin: G.Context(1520598366543).V().Flows().Sort()
12:26:38.581 --- FAIL: TestBridge (31.99s)
```

**Comment by safchain** [comment]
simple.sh is used by some functional tests, please check if the changes don't break them

**Comment by hunchback** [comment]
I made a point of not breaking the existing API, will check.

**Comment by hunchback** [comment]
I see no correlation between changes to `scripts.sh` and the failure:

```
12:26:31.626 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:32.667 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:33.716 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:34.756 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:35.797 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:36.845 Gremlin: g.Context(1520598366543).V().Has("Type", "bridge", "Name", "br-test").Out("Name", "intf1")
12:26:37.248 2018-03-09T13:26:37.183+0100	DEBUG	flow/table.go:231 (*Table).update	dpdk2: flow table update: 1520598387183, 1520598397183
12:26:37.282 2018-03-09T13:26:37.218+0100	DEBUG	flow/table.go:231 (*Table).update	dpdk2: flow table update: 1520598387217, 1520598397217
12:26:37.892 Gremlin: G.Context(1520598366543)
12:26:38.431 Gremlin: G.Context(1520598366543).V().Flows().Sort()
12:26:38.581 --- FAIL: TestBridge (31.99s)
```

Moreoever - I see no direct relation between `simple.sh` and the tests scripts undr `scripts/ci`

**Comment by safchain** [comment]
rerun tests

**Comment by hunchback** [comment]
@safchain - any additional comments?

**Comment by safchain** [comment]
I would make the bridge stuff optional

**Comment by hunchback** [comment]
@safchain - ok bridge is optional now using `-b` or `--bridge`

**Comment by hunchback** [comment]
@lebauce - can you rerun tests?

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun selenium-tests

File: scripts/simple.sh [hunk 1]
```diff
@@ -2,57 +2,175 @@
...
+	fi
+
+	sudo ip netns del $vm
+}
+
+function start() {
 	set -x
+	sudo ovs-vsctl add-br br-int
+	vm_start $NS1 $SUBNET1 $DELAY2 $DELAY1
+	vm_start $NS2 $SUBNET2 $DELAY3 $DELAY4
```
[the remark this answers is not in the document]
Comment (hunchback) [reply scripts/simple.sh]: it still is optional:

```
function vm_start() {
        local vm=$1
        local ip=$2
        [ -n "$ip" ] || return
        ...
}
```

File: scripts/simple.sh [hunk 2]
```diff
@@ -2,57 +2,175 @@
...
 
-	start $2 $3
-else
-	stop
-fi
+case $1 in
+	start)
+		if [ -n "$2" ]; then
+			SUBNET1=$1
+			SUBNET2=$2
```
[the remark this answers is not in the document]
Comment (hunchback) [reply scripts/simple.sh]: changed to:
```
SUBNET1=$2
SUBNET2=$3
```

File: scripts/simple.sh [hunk 3]
```diff
@@ -2,57 +2,175 @@
...
+set -- "${POSITIONAL[@]}" # restore positional parameters
 
-	start $2 $3
-else
-	stop
-fi
+case $1 in
+	start)
+		if [ -n "$2" ]; then
+			SUBNET1=$1
```
[the remark this answers is not in the document]
Comment (hunchback) [reply scripts/simple.sh]: yes - changed

File: scripts/simple.sh [hunk 4]
```diff
@@ -2,57 +2,175 @@
...
+			SUBNET2=$2
+		fi
+		start
+		;;
+	stop)
+		stop
+		;;
+	ping)
+		if [ -n "$2" ]; then
+			IP1=$1
```
[the remark this answers is not in the document]
Comment (hunchback) [reply scripts/simple.sh]: changed:
```
IP1=$2
IP2=$3
```

File: scripts/simple.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hunchback) [reply scripts/simple.sh]: yes ... it still is optional (when setting SUBNET<X> to $<X+1>)

**Comment by hunchback** [comment]
@lebauce @safchain - I notice we are still only running one test - should I open a new PR?

**Comment by lebauce** [comment]
@hunchback I don't think it's necessary to open a new PR..
We have CI issues right now, no job are scheduled

**Comment by safchain** [comment]
please rebase this PR

**Comment by safchain** [comment]
rerun tests

File: scripts/simple.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (nplanel) [reply scripts/simple.sh]: would be better and more readable to test before the call like and avoid function trick, like
if [ -n "$SUBNET2" ] ; then
   vm_start $NS2 $SUBNET2 $DELAY3 $DELAY4
fi

and remove the trick  [ -n "$ip" ] || return

**Comment by safchain** [comment]
rerun tests

**Pull request merged**